libminifi/src/core/RepositoryFactory.cpp

```diff
@@ -96,6 +96,62 @@ std::unique_ptr<core::ContentRepository> createContentRepository(const std::stri
   throw std::runtime_error("Support for the provided configuration class could not be found");
 }
 
+class NoOpThreadedRepository : public core::ThreadedRepository {
+ public:
+  explicit NoOpThreadedRepository(const std::string& repo_name)
+    : core::SerializableComponent(repo_name),
+    ThreadedRepository(repo_name) {
+  }
+
+  ~NoOpThreadedRepository() override {
+    stop();
+  }
+
+ private:
+  void run() override {
+  }
+
+  std::thread& getThread() override {
+    return thread_;
+  }
+
+  std::thread thread_;
+};
+
+std::unique_ptr<core::ThreadedRepository> createThreadedRepository(const std::string& configuration_class_name, bool fail_safe, const std::string& repo_name) {
+  std::string class_name_lc = configuration_class_name;
+  std::transform(class_name_lc.begin(), class_name_lc.end(), class_name_lc.begin(), ::tolower);
+  try {
+    auto return_obj = core::ClassLoader::getDefaultClassLoader().instantiate<core::ThreadedRepository>(class_name_lc, class_name_lc);
+    if (return_obj) {
+      return_obj->setName(repo_name);
+      return return_obj;
+    }
+    // if the desired repos don't exist, we can try doing string matches and rely on volatile repositories
+    if (class_name_lc == "flowfilerepository" || class_name_lc == "volatileflowfilerepository") {
+      return_obj = instantiate<repository::VolatileFlowFileRepository>(repo_name);
+    } else if (class_name_lc == "provenancerepository" || class_name_lc == "volatileprovenancefilerepository") {
+      return_obj = instantiate<repository::VolatileProvenanceRepository>(repo_name);
+    } else if (class_name_lc == "nooprepository") {
+      return_obj = std::make_unique<core::NoOpThreadedRepository>(repo_name);
+    }
+    if (return_obj) {
+      return return_obj;
+    }
+    if (fail_safe) {
+      return {};
+    } else {
+      throw std::runtime_error("Support for the provided configuration class could not be found");
+    }
+  } catch (const std::runtime_error &) {
+    if (fail_safe) {
+      return {};
+    }
+  }
+
+  throw std::runtime_error("Support for the provided configuration class could not be found");
```

Comment:
We don't need to throw here if we just call a `throw` in the catch block rethrowing the original exception.

Reply:
Done.